sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,500 @@
+has not been called.
+
+== `task_sequence` Properties
+
+The following code and table describe the "balanced", "invocation_capacity", and
+"response_capacity" properties that can be provided when declaring a
+`task_sequence`.
+
+[source,c++,linenums]
+----
+namespace sycl {
+namespace ext {
+namespace intel {
+namespace experimental {
+
+struct balanced_key {
+  using value_t = property_value<balanced_key>;
+};
+
+struct invocation_capacity_key {
+  template <uint32_t Size>
+  using value_t = property_value<invocation_capacity_key,
+    std::integral_constant<uint32_t, Size>>;
+};
+
+struct response_capacity_key {
+  template <uint32_t Size>
+  using value_t = property_value<response_capacity_key,
+    std::integral_constant<uint32_t, Size>>;
+};
+
+inline constexpr balanced_key::value_t balanced;
+
+template <> struct is_property_key<balanced_key> : std::true_type {};
+
+// TODO: Not sure if the template here should just be
+// template<auto &f, class propertiesT>
+template <typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...),
+  class propertiesT>
+struct is_property_key_of<balanced_key,
+  task_sequence<f, propertiesT>> : std::true_type {};
+
+template <uint32_t Size>
+inline constexpr invocation_capacity_key::value_t<Size> invocation_capacity;
+
+template <> struct is_property_key<invocation_capacity_key> : std::true_type {};
+
+template <typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...),
+  class propertiesT>
+struct is_property_key_of<invocation_capacity_key,
+  task_sequence<f, propertiesT>> : std::true_type {};
+
+template <uint32_t Size>
+inline constexpr response_capacity_key::value_t<Size> response_capacity;
+
+template <> struct is_property_key<response_capacity_key> : std::true_type {};
+
+template <typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...),
+  class propertiesT>
+struct is_property_key_of<response_capacity_key,
+  task_sequence<f, propertiesT>> : std::true_type {};
+
+} // namespace experimental
+} // namespace intel
+} // namespace ext
+} // namespace sycl
+----
+
+--
+[options="header"]
+|===
+| Property | Description
+| balanced | The `balanced` property is a guarantee to the SYCL device compiler
+that the `task_sequence` object will call exactly the same number of `async` s
+and `get` s over the object's lifetime (i.e. before the `task_sequence`
+destructor is invoked). 
+
+| invocation_capacity | The `invocation_capacity` property defines the minimum
+required number `Size` of `async` invocations that must be invokable without any
+`get` being called on that same set.  This number of `async` invocations without
+a `get` call is the minimum number that will be supported before the `async`
+member function blocks.
+
+| response_capacity | The `response_capacity` property defines the maximum
+number `Size` of outstanding `async` invocations such that all outstanding
+invocations are guaranteed to make forward progress. Further `async` invocations
```

Comment:
I think the guarantee we can provide the user when they give us a response capacity is weaker than this.  For example, in our FPGA hardware, if an earlier invocations stalls it can block later invocations, thereby preventing them from making forward progress.  The only guarantee we are giving here is that at least one of the outstanding invocations can make progress if the number of outstanding invocations is less than response capacity.  This is analogous to the concept of blocking with forward progress guarantee delegation from the C++ spec.

Reply:
I am now using the wording based on an earlier suggestion of yours:
```
Given a response capacity of `N`, the user is guaranteed that up to `N` outstanding calls can make progress.
```
Is this acceptable or would you prefer we re-word this to say 'at least one is guaranteed to make progress'?

Reply:
At least one

Reply:
Updated